Remove mkdocs config

stevepiercy opened the pull request:
See #3042

stevepiercy commented:
Oops. I missed that it was used in the `README.md`. I searched only within `/docs/source/`, and found no reference to `/docs/logos/`. We should move the directory where it is most useful, perhaps at the root at `/logos/`?

sneridagh commented:
@stevepiercy ok for me!

stevepiercy commented on Makefile:
```diff
@@ -88,8 +83,10 @@ bin/python:
-docs-clean:  ## Clean docs build directory
+docs-clean:  ## Clean docs build directory (and legacy docs installs)
 	cd $(DOCS_DIR) && rm -rf $(BUILDDIR)/
+	rm -rf bin include lib
```
This line should be removed because this virtual environment is now used by Sphinx and MyST. Or we can leave it, but you would need to reinstall Sphinx, MyST, and their dependencies. In another PR `requirements-docs.txt` now has the Sphinx and MyST requirements.

stevepiercy replied:
Actually, I see that it should remain, because we only install requirements when the `bin/python` is missing or has changed. Instead I just updated the help comment in 4fa3b641c37262de2f8352a60fc383c4aad87b7a

stevepiercy commented:
> Also, we should remove this also, right?
> 
> https://github.com/plone/volto/blob/b13658327ff11d4b73a0c999798845b42f7ecf49/docs/source/conf.py#L198

We have an open issue where we are discussing whether to keep or remove the Netlify logo in the main Plone Docs:

https://github.com/plone/documentation/issues/1167

stevepiercy commented:
> Also a leftover from https://github.com/plone/volto/blob/3042-remove-mkdocs-config/.github/workflows/docs.yml#L60

That entire file is obsolete. We need to discuss what, if anything, y'all want to test and build with docs, and whether y'all want to deploy anything, such as previews to Netlify. I would suggest that we at least do the following:

```bash
make docs-linkcheckbroken  # Required.
make docs-html  # Required.
make docs-spellcheck  # Optional, and only locally, otherwise it will never pass.
```

sneridagh commented:
@stevepiercy this is ready, right?

stevepiercy commented:
> @stevepiercy this is ready, right?

Not yet. See [previous comment](https://github.com/plone/volto/pull/3070#issuecomment-1042375884).

If you provide direction, I can implement it.

sneridagh commented:
> That entire file is obsolete.

Agreed, we have to update it to the new workflow (I'd use the same approach you did with Netlify for PRs). 
We would need to add StoryBook build to the main docs too, but that should go to a PR in main docs repo.

stevepiercy commented:
Updated GitHub workflow in https://github.com/plone/volto/pull/3070/commits/e07ee6b45567a8ab0a4bfe8a66aa1d18119dc8da.

Please review. Feedback welcome.

As far as Netlify, it appears I do not have sufficient access on the Volto repo to add it to Netlify. I can add  members to the team on Netlify when given an email address, and I can walk y'all through the configuration.

sneridagh commented:
@stevepiercy what should I do to enable it?

stevepiercy commented:
After I sent my request to authorize Netlify for access to `plone/volto`, I think it sends an email to an admin for the Plone GitHub organization. @tkimnguyen let me know he got a request to authorize, and he can provide details. Today I was able to add `plone/volto` as a Netlify site, and I added its configuration and updated `Makefile`.

Building is slow because Netlify detects .nvmrc and yarn installs everything before it starts on Python and Sphinx. Silly Netlify. 🤪 

I'm fiddling with the configuration, and I should figure out the proper knobs to tune.

stevepiercy commented:
w00t! We got Netlify preview working: https://620f6520bdbe750008a828d1--volto.netlify.app/

nileshgulia1 commented:
@stevepiercy uuuuu nicee..

nileshgulia1 reviewed (approved):
This can be merged I guess @sneridagh
